File: DART/alzheimer/app1/app/src/main/java/com/example/app1/MainActivity.java
package com.example.app1;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.*;

public class MainActivity extends AppCompatActivity {

    private TextView mtextques;
    private Button falsebut;
    private Button truebut;
    private int mquesindex=0;
    private int mquizequestion;
    private Quizemodel[] quesarr =new Quizemodel[]{
            new Quizemodel(R.string.q1,true),
            new Quizemodel(R.string.q2,false),
            new Quizemodel(R.string.q3,true),
            new Quizemodel(R.string.q4,true),
            new Quizemodel(R.string.q5,false),
            new Quizemodel(R.string.q6,false),
            new Quizemodel(R.string.q7,true),
            new Quizemodel(R.string.q8,false),
            new Quizemodel(R.string.q9,true),
            new Quizemodel(R.string.q10,true),


    };


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mtextques=findViewById(R.id.txtq);
        Quizemodel q1=quesarr[mquesindex];

        mquizequestion=q1.getMque();
        mtextques.setText(mquizequestion);


        truebut=findViewById(R.id.truebut);
        View.OnClickListener myclicklistner= new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                evaluat(true);
                changequest();

            }
        };
        truebut.setOnClickListener(myclicklistner);
        falsebut =findViewById(R.id.falsebut);
        View.OnClickListener falselistner= new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                changequest();
                evaluat(false);

            }
        };
        falsebut.setOnClickListener(falselistner);

    }



    private void changequest()
    {
        mquesindex=(mquesindex+1)%10;
        mquizequestion=quesarr[mquesindex].getMque();
        mtextques.setText(mquizequestion);
    }


    private void evaluat(boolean usergues)
    {
        boolean currentQuestion=quesarr[mquesindex].ismAnswer();
        if(currentQuestion==usergues)
        {
            Toast.makeText(getApplicationContext(),R.string.correct_toast_message,Toast.LENGTH_SHORT).show();
        }
        else
        {
            Toast.makeText(getApplicationContext(),R.string.incorrecte_toast_message,Toast.LENGTH_SHORT).show();
        }

    }


}

File: kotlin/mems/app/src/main/java/fritprfojects/example/mems/MainActivity.kt
package fritprfojects.example.mems

import android.content.Intent
import android.graphics.drawable.Drawable
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Toast
import com.android.volley.Request
import com.android.volley.Response
import com.android.volley.toolbox.JsonObjectRequest
import com.android.volley.toolbox.StringRequest
import com.android.volley.toolbox.Volley
import com.bumptech.glide.Glide
import com.bumptech.glide.load.DataSource
import com.bumptech.glide.load.engine.GlideException
import com.bumptech.glide.request.RequestListener
import com.bumptech.glide.request.target.Target
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    var currentImageUrl:String?= null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        //code


        fun loadData() {
            Progress.visibility= View.VISIBLE
            val queue = Volley.newRequestQueue(this)
            val url = "https://meme-api.herokuapp.com/gimme"

            //request a tring respose from proviedes
            val jsonObjectRequest = JsonObjectRequest(
                Request.Method.GET, url, null,
                Response.Listener { response ->
                    currentImageUrl=response.getString("url")

                    Glide.with(this).load(currentImageUrl).listener(object : RequestListener<Drawable>{

                        override fun onLoadFailed(
                            e: GlideException?,
                            model: Any?,
                            target: Target<Drawable>?,
                            isFirstResource: Boolean
                        ): Boolean {
                            Progress.visibility =View.GONE
                           return false
                        }


                        override fun onResourceReady(
                            resource: Drawable?,
                            model: Any?,
                            target: Target<Drawable>?,
                            dataSource: DataSource?,
                            isFirstResource: Boolean
                        ): Boolean {
                            Progress.visibility =View.GONE
                           return false
                        }
                    }).into(memeimg)

                },
                Response.ErrorListener {
                    Toast.makeText(this,"Something went wrong ",Toast.LENGTH_SHORT).show()
                })
            queue.add(jsonObjectRequest)

        }

        Sharebtn.setOnClickListener{
            val intent= Intent(Intent.ACTION_SEND)
            intent.type = "text/plain"
            intent.putExtra(Intent.EXTRA_TEXT,"Hey! check this meme $currentImageUrl")
            val chooser=Intent.createChooser(intent,"Share meme")
            startActivity(chooser)

        }

        Nextbtn.setOnClickListener{
            loadData()
        }
}
}